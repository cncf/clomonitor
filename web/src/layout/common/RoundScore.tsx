import getCategoryColor from '../../utils/getCategoryColor';
import styles from './RoundScore.module.css';

interface Props {
  score: number;
  className?: string;
}

const RoundScore = (props: Props) => {
  const color = getCategoryColor(props.score);

  return (
    <div
      data-testid="global-score"
      className={`d-flex align-items-center justify-content-center rounded-pill ${styles.score} ${props.className}`}
    >
      <div className={styles.chart}>
        <svg viewBox="0 0 36 36" className="d-block">
          <path
            className={styles.circleBg}
            d="M18 2.0845
          a 15.9155 15.9155 0 0 1 0 31.831
          a 15.9155 15.9155 0 0 1 0 -31.831"
          />
          <path
            className={styles.circle}
            strokeDasharray={`${props.score}, 100`}
            style={{ stroke: `var(--rm-${color})` }}
            d="M18 2.0845
          a 15.9155 15.9155 0 0 1 0 31.831
          a 15.9155 15.9155 0 0 1 0 -31.831"
          />
          <text x="18" y="23" className={styles.value}>
            {props.score}
          </text>
        </svg>
      </div>
    </div>
  );
};

export default RoundScore;
