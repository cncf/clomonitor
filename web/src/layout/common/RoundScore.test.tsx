import { render, screen } from '@testing-library/react';

import RoundScore from './RoundScore';

describe('RoundScore', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<RoundScore score={80} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<RoundScore score={80} />);

    expect(screen.getByTestId('global-score')).toBeInTheDocument();
    expect(screen.getByText('80')).toBeInTheDocument();
  });
});
