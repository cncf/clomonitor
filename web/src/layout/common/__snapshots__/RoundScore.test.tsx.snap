// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundScore creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-center rounded-pill score undefined"
    data-testid="global-score"
  >
    <div
      class="chart"
    >
      <svg
        class="d-block"
        viewBox="0 0 36 36"
      >
        <path
          class="circleBg"
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <path
          class="circle"
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          stroke-dasharray="80, 100"
          style="stroke: var(--rm-green);"
        />
        <text
          class="value"
          x="18"
          y="23"
        >
          80
        </text>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
